hw3/01-characters.js

```diff
@@ -1,4 +1,58 @@
 // url for the Thrones API
 const url = 'https://thronesapi.com/api/v2/Characters';
 
-fetch(url);
+// async function to fetch the data from a given url
+const fetchData = async function fetchDataFromApi(url) {
+    try {
+        const res = await fetch(url);
+        const data = await res.json();
+        return data;
+    }
+    catch {
+        console.error('Error', error);
+    }
+};
+
+// function to create character cards and add them populate in the DOM from the given data.
+const serveCards = function createAndBuildCardDeck(data) {
+    data.forEach(element => {
+
+        const newCard = document.createElement('div');
+        newCard.className = 'flex-item';
+
+        const imageContainer = document.createElement('div');
+        imageContainer.className = 'card-image';
+
+        const cardImage = document.createElement('img');
+        cardImage.src = element.imageUrl;
+        cardImage.className = "card-image";
+        cardImage.alt = `An image of the character ${element.firstName} ${element.lastName} from the series, Game of Thrones.`;
+        
+        imageContainer.appendChild(cardImage);
+        newCard.appendChild(imageContainer);
+
+        const cardName = document.createElement('h2');
+        cardName.innerText = `${element.firstName} ${element.lastName}`;
+
+        newCard.appendChild(cardName);
+
+        const cardInfo = document.createElement('p');
+        cardInfo.innerText = `${element.title}`
+
+        newCard.appendChild(cardInfo);
+
+        document.getElementById("charSection").appendChild(newCard);
+    });
+};
+
+//async main function used for wrapping other async functions.
+const main = async function main() {
+
+    let characterArray = [];
+    characterArray = await fetchData(url);
```

Comment:
An `await` expression will wait for the function to complete, so there's no reason to split up the declaration and assignment.

This can just be `const characterArray = await fetchData(url);`.